aea/cli_gui/__init__.py

```diff
@@ -572,7 +581,29 @@ def get_process_status(process_id: subprocess.Popen) -> ProcessState:
 
 def _kill_running_oef_nodes():
     logging.info("Kill off any existing OEF nodes which are running...")
-    _call_subprocess(["docker", "kill", oef_node_name], timeout=30.0)
+    # find already running images
+    image_ids = set()
+
+    process = subprocess.Popen(
+        ["docker", "ps", "-q", "--filter", "ancestor=fetchai/oef-search:0.7"],
+        stdout=subprocess.PIPE,
+    )
+    process.wait(2.0)
+    (stdout, stderr) = process.communicate()
+    image_ids.update(stdout.decode("utf-8").splitlines())
+
+    process = subprocess.Popen(
+        ["docker", "ps", "-q", "--filter", "name=" + oef_node_name],
+        stdout=subprocess.PIPE,
+    )
+    process.wait(2.0)
+    (stdout, stderr) = process.communicate()
+    image_ids.update(stdout.decode("utf-8").splitlines())
+
+    if stdout != b"":
+        _call_subprocess(
+            ["docker", "kill", *list(image_ids)], timeout=30.0, stdout=subprocess.PIPE
+        )
```

Comment:
Should there be another process finally block here?

Reply:
it is done inside `_call_subprocess)`